anime_downloader/extractors/gcloud.py

```diff
@@ -10,22 +10,34 @@
 class Gcloud(BaseExtractor):
     def _get_data(self):
         url = self.url
-        
-        """gcloud uses the same video ID as other sites"""
-        url = url.replace('fembed.com','gcloud.live')
-        url = url.replace('feurl.com','gcloud.live')
-        
-        url = url.replace('gcloud.live/v/','gcloud.live/api/source/')
-        if url.find('#') != -1:url = url[:url.find('#')] 
-        url = (url[-url[::-1].find('/'):])
-        data = helpers.post(f'https://gcloud.live/api/source/{url}').json()['data']
 
+        # TODO: Maybe put it in Option ?
+        withPreserveServerPreference = True
```

Comment:
Instead of bodging this with an option (which I doubt works on fembed and feurl) I'd advise making it independent extractors.

Reply:
The problem, which may not be one, is that the gcloud extractor use each time the server gcloud.live.
i tried to use the orginal server like:

```
gcloud.live/v/... -> gcloud.live/api/source/..
www.fembed.net/v/.. -> www.fembed.net/api/source/ (and not www.gcloud.live/api/source/...)
```

However the old code of gcloud ( with withPreserveServerPreference = False) not work for my glcoud url server list like :
```
https://www.mg-play.info/v/...
https://youtubedownloader.cx/v/...
```

when you use (withPreserveServerPreference = True)  the rule is :
`https://server/v/... -> https://server/api/source/...`

But i could create a new extractor as gcloud2.py but i think it is not a good idea to have 2 extactor for the same aim

Reply:
Yes I could understand the code, but what I'm saying is that it does not fit in gcloud extractor. Multiple extractors running (essentially) the same code is good because:
1) Way easier (and cleaner) to fix issues with one of the sites 
2) Backwards compatibility gets better, no need to go back and change all references to the extractor in old code
3) Easier for newcomers, who would guess fembed links works in gcloud extractor?
I'm not asking for gcloud2.py, rather fembed.py, the reason that's not already in the project is just laziness (as mentioned before). 
`withPreserveServerPreference` will fail if the url is fembed.com as their api isn't the same as gcloud and feurl.

Reply:
ok but you can get a server url like glcoud.live, gcloud.com, gcloud.eu, ...
i think is the role of the site to choose the good extractor like in my code the function : [FindBetterExtractor](https://github.com/BlackDady/anime-downloader/blob/master/anime_downloader/sites/animesvostfr.py)

if we add it on util.py and complete this function for choose extractor, for newcomers it can be cool.
for me it would have been more cool

Reply:
Findbetterextractor can definitely be moved to utils and I do see the use for it, but changing core stuff is not up to me, I suggest you make it a separate PR and ask @vn-ki for a review.

Reply:
>  glcoud.live, gcloud.com, gcloud.eu, ...

The points above should theoretically apply to these, but I'm more lenient with conflicting urls if they share the same API

Reply:
i think i will cancel my pull request because i think this project is old and dead. It was just for funny that i look it to improve my python level. you are the only member actif of this project
youtube-dl seems have a good community and a lot of extractor. Maybe make a client for anime with youtube-dl embeded api could be more actuality ? no ?